### DISCUSSION TITLE
Cannot load glb files from firebase bucket storage

### DISCUSSION THREAD

**Discussion by nirajmohanrana** [google/model-viewer#4158] [Q&A] [answered]
### Cannot load glb files from firebase bucket storage
After fetching the glb file from the firebase bucket, it shows this error that 
![image](https://user-images.githubusercontent.com/71956273/224617987-a90ccd83-e95f-4c5f-b3d7-6727a271241e.png)

The files are of glTF versions 2.0, then too it shows this error

I have stored an object which also contains link to glb file "product.glbFile"
which axios to fetch the file
![image](https://user-images.githubusercontent.com/71956273/224618400-1a7bdec9-3cf0-45b8-950e-094fd7e00d2c.png)
I am working on a react js framework

Can someone please help me to solve this issue

I have tried this https://github.com/google/model-viewer/discussions/3511 but it did not worked for me

**Comment by elalish** [comment]
Please load the file in our editor, which runs the glTF validator and see if it produces any errors or warnings.

**Reply by nirajmohanrana** [reply]
There in no errors or warning in [Model Editor](https://modelviewer.dev/editor/)
![image](https://user-images.githubusercontent.com/71956273/225813544-3ddb6f82-fb6b-419b-b66a-f1182276cc0b.png)
Here the Validation Report from [GLTF Viewer](https://gltf-viewer.donmccurdy.com/)

I need to fix this ASAP, please do check it.

**Reply by elalish** [reply]
If it loads in the editor, it should load in MV (since the editor uses MV). What MV version are you using?

**Reply by nirajmohanrana** [reply]
I am using version 3.0.1

**Reply by nirajmohanrana** [reply]
Is it problem related to firebase? Is there any other cloud file storage I can use.
(which Cloud Storage is used by model [modelviewer.dev](https://modelviewer.dev/)?)

**Reply by elalish** [reply]
modelviewer.dev is a Github Pages site, so everything is served from there. Any cloud storage where you can set CORS headers properly should work, see: https://modelviewer.dev/docs/faq.html#entrydocs-technical-questions-cors

**Reply by elalish** [reply]
I would guess the problem is in the axios fetching code.

**Reply by nirajmohanrana** [reply]
I have editted it the CORS Headers in firebase, but it shows same error.
How about I convert the files to dataString and then parse it "src",I tried it locally it runs perfectly fine, is it a Good Idea, what your views?? or is there any drawbacks to it.

![image](https://user-images.githubusercontent.com/71956273/226554810-65dd268f-8747-40e4-bc7e-1d13890fb61a.png)
![image](https://user-images.githubusercontent.com/71956273/226554656-e8b36369-f5e8-4788-97bc-dd55101096f8.png)

**Reply by elalish** [reply]
That's odd, but if it works, it works. I think you might hit a size limit at some point, but other than that it's fine. You can also pass in an ObjectURL wrapping a `Blob`.

**Comment by nirajmohanrana** [comment, marked as answer]
Got the answer for [this issue](https://github.com/google/model-viewer/discussions/4158#discussion-4956901), actually Firebase don't allow to fetch files directly (obviously for security reasons, except native web files like images). To access GLB files we need to change CORS header, which are not just rules that can be changed in Firebase>Rules.

**Here is the steps to Change CORS Headers**:
Step1: Create cors.json file:
`[
    {
      "origin": ["https://your-website.com"],
      "method": ["GET"],
      "maxAgeSeconds": 3600
    }
]`

....here in origin add the url of your site from where request is sent, to accept it from every site add this `"origin": ["*"],`

Step2: Download [GSUTIL](https://dl.google.com/dl/cloudsdk/channels/rapid/GoogleCloudSDKInstaller.exe) from here
You can follow this steps [gsutil installation guide](https://cloud.google.com/storage/docs/gsutil_install)

Step3: After intsallation open gCloud command line and run `gcloud init` and login to your firebase google account.

Step4: cd to the location of cors.json file and run `gsutil cors set cors.json gs://BUCKET_NAME`
change BUCKET_NAME to your app bucket

Step5: Your CORS Header setting will be chnaged to confirm it run `gsutil cors get gs://BUCKET_NAME`

Now your are good to go and fetch files as needed, @elalish thanks for your support
